perf-tool/src/server.cpp

```diff
@@ -109,133 +143,178 @@ void Server::handleServer()
-                sendMessage(newsocketFd, "Connection to server succeeded", "ServerStartEvent", "Server");
-
-                /* Update number of active clients */
-                activeNetworkClients++;
+                /* First check for incoming commands from network clients */
+                for (int i = 0; i < activeNetworkClients; i++)
+                {
+                    int socketIndex = ServerConstants::BASE_POLLS + activeNetworkClients;
+                    if (pollFds[socketIndex].revents & (POLLERR | POLLHUP | POLLNVAL))
+                    {
+                        /* Error encountered; we should close the socket */
+                        if (pollFds[socketIndex].fd > 0)
+                        {
+                            networkClients[i]->closeFd();
+                            pollFds[socketIndex].fd = -1; /* TODO: we need a linked list rather than an array */
+                        }
+                    }
+                    else if (pollFds[ServerConstants::BASE_POLLS + activeNetworkClients].revents & POLLIN)
+                    {
+                        command = networkClients[i]->handlePoll();
+                        if (!command.empty())
+                        {
+                            handleClientCommand(command, "Client");
+                        }
+                    }
+                }
+                /* Now, handle new connection requests */
+                if (pollFds[0].revents & POLLIN)
+                {
+                    /* The accept() call actually accepts an incoming connection */
+                    clilen = sizeof(cli_addr);
+
+                    /* This accept() function will write the connecting client's address info 
+                     * into the the address structure and the size of that structure is clilen.
+                     */
+                    newsocketFd = accept(serverSocketFd, (struct sockaddr *)&cli_addr, &clilen);
+                    if (newsocketFd < 0)
+                    {
+                        perror("ERROR on accept");
+                    }
+                    else
+                    {
+                        /* Check the upper limit on active connections */
+                        if (activeNetworkClients < ServerConstants::NUM_CLIENTS)
+                        {
+                            networkClients[activeNetworkClients] = new NetworkClient(newsocketFd);
+                            if (verbose >= Verbose::INFO)
+                                printf("server: got connection from %s port %d\n",
+                                        inet_ntoa(cli_addr.sin_addr),
+                                        ntohs(cli_addr.sin_port));
+
+                            /* Send a welcome message */
+                            sendMessage(newsocketFd, "Connection to server succeeded", "ServerStartEvent", "Server");
+
+                            /* Update number of active clients */
+                            activeNetworkClients++;
```

Comment:
This may be me not understanding how `poll` works, but don't we need to initialize the `socketIndex`th entry of the `pollFds` array here?

Reply:
You are correct. The original code didn't have it, but it worked because it attempted a read operation on all sockets, regardless of whether there was any data sent by the client.
Fixed it.